packages/react-lowcode/src/codegen/generation/generators/detail/mui-detail-generator.ts

```diff
@@ -48,6 +51,155 @@ export default class MuiDetailGenerator implements DetailGenerator {
     );
   }
 
+  async getFormWidgetProperties(position: SourceLineCol): Promise<WidgetProperties> {
+    let result: WidgetProperties = {
+      properties: []
+    };
+
+    if (this._widgetContext) {
+      const sourceCode = await this._widgetContext.getSourceCodeString(position);
+      const ast = createAst(sourceCode);
+
+      if (ast) {
+        result = this.getFormWidgetPropertiesFromAst(ast, position);
+      }
+    }
+
+    return result;
+  }
+
+  getFormWidgetPropertiesFromAst(ast: ts.SourceFile, position: SourceLineCol): WidgetProperties {
+    const result: WidgetProperties = {
+      properties: []
+    };
+
+    const pos = ast.getPositionOfLineAndCharacter(position.lineNumber - 1, position.columnNumber - 1);
+    const element = findByCondition<Node>(ast, (node: Node) => {
+      return node.pos === pos;
+    });
+
+    if (element) {
+      if (ts.isJsxOpeningElement(element) || ts.isJsxSelfClosingElement(element)) {
+        element.attributes.properties.forEach(prop => {
+          if (ts.isJsxAttribute(prop)) {
+            const propName = prop.name.escapedText.toString();
+
+            if (prop.initializer) {
```

Comment:
please do not to this nested `if`s (it is not a mainstream... and it is nested too many times)

Reply:
Now it's ok for the "getFormWidgetProperties" function. I still need to edit the "setFormWidgetProperties" function